### PR TITLE
all: log picker when updating balancing state

### PR DESCRIPTION
**Pull request by dapengzhang0**
Let `ManagedChannelImpl` log picker update when updating balancing state. (https://github.com/grpc/grpc-java/compare/master...dapengzhang0:log-picker?expand=1#diff-354509714620f5c493a6810aab9419f2R1156)

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java [hunk 1]
```diff
@@ -431,6 +431,11 @@ public String toString() {
     public PickResult pickSubchannel(PickSubchannelArgs args) {
       return PickResult.withNoResult();
     }
+
+    @Override
+    public String toString() {
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java]: The default toString() should work fine, or toStringHelper.

File: core/src/main/java/io/grpc/internal/OobChannel.java [hunk 2]
```diff
@@ -180,6 +180,11 @@ public Object getInternalSubchannel() {
         public PickResult pickSubchannel(PickSubchannelArgs args) {
           return result;
         }
+
+        @Override
+        public String toString() {
+          return "Subchannel picker with pick result: " + result;
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/internal/OobChannel.java]: Better to use toStringHelper in all cases to produce standardized outputs.

File: core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java [hunk 3]
```diff
@@ -63,6 +63,11 @@ public void handleNameResolutionError(final Status error) {
             public PickResult pickSubchannel(PickSubchannelArgs args) {
               return PickResult.withError(error);
             }
+
+            @Override
+            public String toString() {
+              return "Error picker with status: " + error;
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java]: ditto

File: core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java (hunk 1 above)
Comment (dapengzhang0) [reply core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java]: Done.

File: core/src/main/java/io/grpc/internal/OobChannel.java (hunk 2 above)
Comment (dapengzhang0) [reply core/src/main/java/io/grpc/internal/OobChannel.java]: Done.

File: core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java (hunk 3 above)
Comment (dapengzhang0) [reply core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java]: Done.

**Pull request merged**